config/initializers/omniauth.rb

```diff
@@ -87,7 +87,13 @@ def call(env)
         response.redirect('/dfe/sessions/new')
         response.finish
       else
-        @app.call(env)
+        begin
+          @app.call(env)
+        rescue ActionController::InvalidAuthenticityToken
+          response = Rack::Response.new
+          response.redirect('/signin')
+          response.finish
```

Comment:
I'd be tempted to remove the repetition in generating a redirect here by creating a `redirect_to` method, maybe I'm used to Code Climate chirping up about it.